langchain/chains/query_constructor/base.py

```diff
@@ -0,0 +1,126 @@
+""""""
+import json
+from typing import Callable, List, Optional, Any
+
+from pydantic import BaseModel
+
+from langchain import BasePromptTemplate, FewShotPromptTemplate, LLMChain
+from langchain.chains.query_constructor.parser import get_parser
+from langchain.chains.query_constructor.prompt import (
+    default_examples,
+    default_prefix,
+    default_schema,
+    default_suffix,
+    example_prompt,
+)
+from langchain.chains.query_constructor.query_language import (
+    Comparator,
+    Operator,
+    StructuredQuery,
+)
+from langchain.output_parsers.structured import parse_json_markdown
+from langchain.schema import BaseLanguageModel, BaseOutputParser, OutputParserException
+
+
+class StructuredQueryOutputParser(BaseOutputParser[StructuredQuery]):
+    ast_parse: Callable
+    """"""
+
+    def parse(self, text: str) -> StructuredQuery:
+        try:
+            expected_keys = ["query", "filter"]
+            parsed = parse_json_markdown(text, expected_keys)
```

Comment:
If the output is modified to be something like

```
<command>
request(..., ...)
</command>
```

Then after extracting the content of `command, the string can be fed into `ast_parse` without having to deal with parsing JSON and dealing with query and filter separately